--- a/src/vnet/ipsec/ipsec_types_api.c
+++ b/src/vnet/ipsec/ipsec_types_api.c
@@ -44,7 +44,7 @@ ipsec_proto_encode (ipsec_protocol_t p)
     case IPSEC_PROTOCOL_AH:
       return clib_host_to_net_u32 (IPSEC_API_PROTO_AH);
     }
-  return (VNET_API_ERROR_UNIMPLEMENTED);
+  return (vl_api_ipsec_proto_t)VNET_API_ERROR_UNIMPLEMENTED;
 }
 
 int
@@ -77,7 +77,7 @@ ipsec_crypto_algo_encode (ipsec_crypto_a
       break;
     }
   ASSERT (0);
-  return (VNET_API_ERROR_UNIMPLEMENTED);
+  return (vl_api_ipsec_crypto_alg_t)VNET_API_ERROR_UNIMPLEMENTED;
 }
 
 int
@@ -109,7 +109,7 @@ ipsec_integ_algo_encode (ipsec_integ_alg
       break;
     }
   ASSERT (0);
-  return (VNET_API_ERROR_UNIMPLEMENTED);
+  return (vl_api_ipsec_integ_alg_t)VNET_API_ERROR_UNIMPLEMENTED;
 }
 
 void
